pkg/scheduler/factory.go

```diff
@@ -343,10 +343,13 @@ func MakeDefaultErrorFunc(client clientset.Interface, podLister corelisters.PodL
 			klog.Warningf("Pod %v/%v doesn't exist in informer cache: %v", pod.Namespace, pod.Name, err)
 			return
 		}
-		// As <cachedPod> is from SharedInformer, we need to do a DeepCopy() here.
-		podInfo.Pod = cachedPod.DeepCopy()
-		if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podQueue.SchedulingCycle()); err != nil {
-			klog.Error(err)
+
+		if len(cachedPod.Spec.NodeName) == 0 {
```

Comment:
Add a comment explaining when the pod may have been assigned spec.nodeName.

Reply:
I added a comment to explain why this check should be added, and refactored the code as you suggested earlier, and I also thought it would be better to add a warning log.